packages/react-utilities/src/compose/makeMergeProps.ts

```diff
@@ -2,12 +2,12 @@ import * as React from 'react';
 import { css } from '@fluentui/utilities';
 import { GenericDictionary } from './types';
 
-export type MergePropsOptions = {
+export type MergePropsOptions<TState = GenericDictionary> = {
```

Comment:
Could we omit `GenericDictionary` altogether ? I think the only consequence would be forcing users to type merge props which isn't a bad thing

```suggestion
export type MergePropsOptions<TState> = {
```

Reply:
It seems to me the only reason that type exists is to no add eslint rules everywhere for `any` usage

Reply:
Yeah I'd like to get rid of `GenericDictionary` from `mergeProps` entirely, but that requires some Typescript-fu and/or refactoring `mergeProps` significantly, which is out of scope of this RFC.

In this case, this type argument is always supplied by `makeMergeProps`, so a default is not necessary.